### ISSUE TITLE
[FEATURE REQUEST] Add custom icons for Layers and Tags on GameObjects. Add Icons for VRCSceneDescriptor and UdonBehaviours.

### ISSUE THREAD

**Issue by BluWizard10** [Dreadrith/HierarchyPlus#1]
Hi there,

I am coming from using QHierarchy and since it may have started to break in Unity 2022, there are some features on it that I feel are missing in your HierarchyPlus system. Some of which are customizable Layers and Tag Icons. In addition, I would like to see Icons for components specifically found in the Worlds SDK, such as the VRCSceneDescriptor and UdonBehaviour.

I would like to know if adding these to HierarchyPlus is possible. If so, that would be awesome.

Thanks!

**Comment by JustBuddy** [comment]
[878b98e](https://github.com/Dreadrith/HierarchyPlus/pull/2/commits/878b98efc55008651e8138e1f728de60c82ca7ba)

**Comment by JustBuddy** [comment]
@Dreadrith I'd say we have most requested features in, what would be missing?
The tag icons are customizable by themselves and already implemented (the gems I refered to in #6).
Custom Icons for WorldSDK are also implemented via #2.
Not sure what customizable layers refers to.

**Comment by BluWizard10** [comment]
> @Dreadrith I'd say we have most requested features in now, what would be missing?
> 
> The tag icons are customizable by themselves and already implemented (the gems I refered to in #6).
> 
> Custom Icons for WorldSDK are also implemented via #2.
> 
> Not sure what customizable layers refers to.

I'm referring to icons that represent the Layer the GameObject is on. Layers are a system in Unity that VRChat heavily depends on in World Creation.

However, now that HierarchyPlus is Open Source and is no longer an obfuscated script (THANK YOU!!), I may be able to add that as an option.

**Comment by Dreadrith** [comment]
Yep, I have this open because I wanted that implemented for those that want it. Getting around to it is a different subject lol.
If you add it yourself, that'd be great help and very much appreciated! 👍

**Comment by Dreadrith** [comment]
I took some time while adding the VRC Constraints icons to add labels for GameObject's Tag and Layer. Available as of [v1.3.0](https://github.com/Dreadrith/HierarchyPlus/releases/tag/v1.3.0).
No indicator for static GOs yet though.
